### PR TITLE
Sales Log Table

### PR DESCRIPTION
**Pull request by hyunzeekim**
## Description

Link to ticket: https://www.notion.so/uwblueprintexecs/Building-Up-be1b30c321554bffba37a64c6cdc1ec6?p=39f075d988f1449a891e8b1107aae51f

## Approach

- Updated `latestOrders` query in https://github.com/uwblueprint/building-up-server/pull/54

## Testing

To test, insert orders in by adding a new `Orders` mutation that adds orders

## Screenshots

sales log table

### CODE REVIEW

File: src/pages/Dashboard/DashboardTeam/SalesLogTable.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      data?.getLatestOrders?.map(order => ({
+        date: new Date(parseInt(order.createdAt, 10)).toDateString(),
+        order: order.orderNumber,
+        qty: order.numberOfItems,
+        price: '$' + order.price.toFixed(2),
+      })) ?? [],
+    [data],
+  );
+
+  const columns = useMemo(
```
Comment (ZhangCreations) [inline src/pages/Dashboard/DashboardTeam/SalesLogTable.js:20]: Does this need to be wrapped in a memo?

File: src/pages/Dashboard/DashboardTeam/SalesLogTable.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        accessor: 'order',
+      },
+      {
+        Header: 'Qty',
+        accessor: 'qty',
+      },
+      {
+        Header: 'Amount Raised',
+        accessor: 'price',
+      },
```
Comment (jzxhuang) [inline src/pages/Dashboard/DashboardTeam/SalesLogTable.js:33]: product/design request: we'd like an additional column for the donation amount, named 'Additional Donation'.

Thanks!

https://uwblueprint.slack.com/archives/C01PVF15KJR/p1619543261000100?thread_ts=1619401525.001300&cid=C01PVF15KJR

File: src/pages/Dashboard/DashboardTeam/SalesLogTable.js (hunk 1 above)
Comment (jzxhuang) [reply src/pages/Dashboard/DashboardTeam/SalesLogTable.js:20]: Interesting question! I did some research: https://react-table.tanstack.com/docs/quick-start

> It's important that we're using React.useMemo here to ensure that our data isn't recreated on every render. If we didn't use React.useMemo, the table would think it was receiving new data on every render and attempt to recalculate a lot of logic every single time. Not cool!

u can look into `useMemo` if you'dl like, it's kinda complicated

File: src/pages/Dashboard/DashboardTeam/SalesLogTable.js (hunk 2 above)
Comment (hyunzeekim) [reply src/pages/Dashboard/DashboardTeam/SalesLogTable.js:33]: Added addition column for donation amount and removed day of the week 👍

**Review by hongyi-chen — APPROVED** [review]
lgtm!

File: src/pages/Dashboard/DashboardTeam/SalesLogTable.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { Box, Skeleton, Table, Thead, Tbody, Tr, Th, Td, chakra } from '@chakra-ui/react';
+
+const SalesLogTable = ({ loading, data }) => {
+  const sales = useMemo(
+    () =>
+      data?.getLatestOrders?.map(order => ({
+        order: order.orderNumber,
+        qty: order.numberOfItems,
+        price: '$' + order.price.toFixed(2),
+        donation: '$' + order.donationAmount.toFixed(2),
```
Comment (jzxhuang) [inline src/pages/Dashboard/DashboardTeam/SalesLogTable.js]: using [template literal strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) could make these 2 lines cleaner!

`${order.price.toFixed(2)}`

**Review by jzxhuang — APPROVED** [review]
looks great! you can make this minor change & merge

**Pull request merged**